File: benchmarks/tsb/bench_index_rename.ts
/**
 * Benchmark: index_rename — Index.rename changing the index name
 */
import { Index } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const data = Array.from({ length: ROWS }, (_, i) => `key_${i}`);
const idx = new Index(data, "original_name");

for (let i = 0; i < WARMUP; i++) {
  idx.rename("new_name");
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  idx.rename("new_name");
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "index_rename",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_index_rename.py
"""Benchmark: index_rename — Index.rename changing the index name"""
import json
import time
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = [f"key_{i}" for i in range(ROWS)]
idx = pd.Index(data, name="original_name")

for _ in range(WARMUP):
    idx.rename("new_name")

start = time.perf_counter()
for _ in range(ITERATIONS):
    idx.rename("new_name")
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "index_rename",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))
